src/ipcache.h

```diff
@@ -227,7 +227,24 @@ typedef Dns::CachedIps ipcache_addrs; ///< deprecated alias
 typedef void IPH(const ipcache_addrs *, const Dns::LookupDetails &details, void *);
 
 void ipcache_purgelru(void *);
-void ipcache_nbgethostbyname(const char *name, IPH * handler, void *handlerData);
+
+void ipcacheNbgethostbynameInternal(const char *name, IPH * handler, void *handlerData);
+
+template <class HandlerData>
+inline void
+ipcache_nbgethostbyname(const char *name, IPH * handler, HandlerData *handlerData)
+{
+    Assure(cbdataReferenceValid(handlerData->toCbdata()));
```

Comment:
Just FYI: One cannot check cbdata validity of an unlocked object. Here, handlerData is not necessarily locked (because data _owner_ does not lock its own data); IpCacheLookupForwarder constructor locks it later, when initializing receiverData.

Our CI even managed to crash this code, with two functionality tests [failing](https://github.com/measurement-factory/squid/actions/runs/14886385142/job/41806982806). Here is a (presumably the same) failure from my local Daft tests:

```
2025/05/07 15:54:55 kid1| FATAL: assertion failed: cbdata.cc:281: "c->locks > 0"
    current cache_peer probe: 127.0.0.1
```